### PR TITLE
Kestrel with pooling

### PR DESCRIPTION
**Pull request by benaadams**
Regular 3671 MB, 21.1 million objects per 1M requests

regular

Current max pooling 428 MB, 9.3 million objects per 1M requests

pooled

Note the header allocations in regular is a recent change https://github.com/aspnet/KestrelHttpServer/pull/608 as they are no longer by default pooled on a connection.

*Aside:* A connection can be shared by many users behind a reverse forwarder; so if you don't like the "risk of pooling"; probably shouldn't share stuff at the connection level

### CODE REVIEW

File: testapp/PooledKestrel/hosting.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+﻿{
+  "hosting.maxPooledContexts": 256,
+  "kestrel.maxPooledStreams": 256,
+  "kestrel.maxPooledHeaders": 256
+}
```
Comment (SajayAntony) [inline testapp/PooledKestrel/hosting.json]: Adding this to a readme.md would definitely help folks understand what exactly is being pooled in `PooledKestrel`

**Comment by SajayAntony** [comment]
:watch:

File: testapp/PooledKestrel/hosting.json (hunk 1 above)
Comment (benaadams) [inline testapp/PooledKestrel/hosting.json]: Added

**Comment by SajayAntony** [comment]
:shipit:

**Comment by halter73** [comment]
Looks like a good scenario to test. :ship:

**Pull request merged**